src/toast/ops/hwpss_model.py

```diff
@@ -0,0 +1,936 @@
+        central_mag = None
+        if obs.comm_col_rank == 0:
+            all_good = [True for x in all_dets]
+            n_cut = 1
+            while n_cut > 0:
+                n_cut = 0
+                mn = np.mean(all_mag[all_good])
+                std = np.std(all_mag[all_good])
+                for idet, det in enumerate(all_dets):
+                    if not all_good[idet]:
+                        continue
+                    if np.absolute(all_mag[idet] - mn) > self.relcal_cut_sigma * std:
+                        all_good[idet] = False
+                        n_cut += 1
+                    # else:
+                    #     print(
+                    #         f"Keeping {det}: {np.absolute(all_mag[idet] - mn)} !> {self.relcal_cut_sigma * std}"
+                    #     )
+            central_mag = np.mean(all_mag[all_good])
+            all_flags = {
+                x: self.hwp_flag_mask for i, x in enumerate(all_dets) if not all_good[i]
+            }
+        if obs.comm_col is not None:
+            all_flags = obs.comm_col.bcast(all_flags, root=0)
+            central_mag = obs.comm_col.bcast(central_mag, root=0)
+
+        # Every process flags its local detectors
+        det_check = set(dets)
+        local_flags = dict(obs.local_detector_flags)
+        for det, val in all_flags.items():
+            if det in det_check:
+                local_flags[det] |= val
+        obs.update_local_detector_flags(local_flags)
+        local_good = [x for x in dets if x not in all_flags]
+
+        return local_good, central_mag
+
+    def _average_magnitude(self, dets, coeff):
+        mag = dict()
+        if self.time_drift:
+            # 4 values per harmonic, 2f is index 1
+            re_comp = 4 * 1 + 0
+            im_comp = 4 * 1 + 2
+        else:
+            # 2 values per harmonic, 2f is index 1
+            re_comp = 2 * 1 + 0
+            im_comp = 2 * 1 + 1
+        n_chunk = coeff.shape[2]
+        for idet, det in enumerate(dets):
+            ch_mag = list()
+            for ch in range(n_chunk):
+                ch_mag.append(
+                    np.sqrt(
+                        coeff[idet, re_comp, ch] ** 2 + coeff[idet, im_comp, ch] ** 2
+                    )
+                )
+            mag[det] = np.mean(ch_mag)
+        return mag
+
+    def _fit_chunk(
+        self, obs, dets, indx, start, end, sincos, sh_flags, det_flags, reltime, coeff
+    ):
+        log = Logger.get()
+        ch_timer = Timer()
+        ch_timer.start()
+
+        # The sample slice
+        slc = slice(start, end, 1)
+        slc_samps = end - start
+
+        if reltime is None:
+            ch_reltime = None
+        else:
+            ch_reltime = reltime[slc]
+
+        obs_cov = hwpss_compute_coeff_covariance(
+            sincos[slc],
+            sh_flags[slc],
+            comm=obs.comm.comm_group,
+            times=ch_reltime,
+            time_drift=self.time_drift,
+        )
+        if obs_cov is None:
```

Comment:
If one chunk fails to fit, it could just be a section of flagged data. Is the assumption that such chunks should be screened somewhere earlier?

Reply:
That is a good point- currently I exclude "tiny" chunks from the fit, but chunks without a solution could just be skipped in the interpolation.

Reply:
This is now handled more gracefully.  Failure to fit individual chunks just results in the removal of that chunk from the overall solution.